README.md

```diff
@@ -119,6 +119,62 @@ ClientConfig = [{reconnect_cool_down_seconds, 10}],
 ok = brod:start_client([{"localhost", 9092}], brod_client_1, ClientConfig).
 ```
 
+## Handling of connection errors
+
+Usually you would have a supervisor that starts your worker module. In the worker's `start_link` method, you should do two things:
+
+1. Make sure the Brod client you're going to use is running by calling `brod:start_client`. That will start the client under Brod's supervisor, if a client with the given name does not already exist. Afterwards, Brod will automatically recover from connection issues.
+1. Start, e.g., a group subscriber by calling `brod:start_link_group_subscriber`, and return the pid for your supervisor to monitor.
+
+(Note that using `brod:start_link_client` to link the client to your supervisor is deprecated. Using it, a connection loss won't trigger a reconnect, even though the process linked to stays alive.)
```

Comment:
I'm not sure what you have observed. but here is how `brod_client` handles connection restart.
1. When `brod_client` start/restart, it'll try to fetch the broker metadata by looping through all the given endpoints, if ALL fail, it gives up and crash, later `brod_sup` should restart it after a configurable delay.
2. A metadata connection is established to fetch metadata, usually after 10 minutes of idel, kafka will close it, `brod_client` will be 'lazy' on it: only start it again when there is a need to fetch metadata
3. A set of payload connections will be established to produce/fetch messages. If a connection is down, there will be a configurable 'cool-down' seconds of delay before trying to re-establish the connection. However, the re-establish is also 'lazy', it is triggered either by producer trying to send a produce request, or a consumer trying to send a fetch request.

start_link_client is deprecated, however it doesn't imply any difference in brod_client's behaviour when it comes to socket error handling.